pgrx/src/datum/array.rs

```diff
@@ -282,6 +289,120 @@ impl<'a, T: FromDatum> Array<'a, T> {
     }
 }
 
+#[derive(thiserror::Error, Debug, Copy, Clone, Eq, PartialEq)]
+pub enum ArrayError {
+    #[error("Array contains nulls")]
+    ArrayContainsNulls,
+}
+
+impl<'a> Array<'a, f64> {
+    /// Returns a slice of `f64`s which comprise this [`Array`].
+    ///
+    /// # Errors
+    ///
+    /// Returns a [`ArrayError::ArrayContainsNulls`] error if this [`Array`] contains one or more
+    /// SQL "NULL" values.  In this case, you'd likely want to fallback to using [`Array::iter()`].
+    pub fn as_slice(&self) -> Result<&[f64], ArrayError> {
+        if self.contains_nulls() {
```

Comment:
At some point it'd be nice to find a way to optimize `contains_nulls`, but for now this is great.

Reply:
I looked at it and don’t see how. BitSlice doesn’t have a fast “at least one zero/one” short-circuit path.